Container-License-Compliance/containers.md

```diff
@@ -0,0 +1,108 @@
+
+Tarball (Image)
+: A single tar file comprising of all the layers that make up an Image. Generated with a 'docker save'
+
+Tarball (container)
+: A single tar file comprising of all the layers that make up a 'Container', squashed into a single Layer. Generated with a 'docker export'
+
+## More information
+
+There are usually three types of container-related deliveries.
+
+### Dockerfiles
+
+Dockerfiles are text files that describe how to build container images. They do not themselves contain any other software but instruct the building tooling to fetch software and/or manipulate the existing configuration to produce an image.
+
+The following example Dockerfile would produce a Clear Linux based Linux image with the git tool installed:
+
+FROM clearlinux:base
+RUN swupd update && swupd bundle-add git
+
+### Container Image (layers)
+
+The container image layers composited from a Dockerfile. The image is composed of one or more layers, and of particular interest is what layers you have added, and which, if any, layers (images) you have based your Image upon (the FROMs in your Dockerfile).
+
+### Container tarfile Images
+
+These are generated from a docker 'export' or 'save'. A full disk image of a container including all the components (layers).  Not to be confused with the executable container images produced locally by the tooling for local container execution.
+
+
+## Things to be taken care of, in order to get approval to deliver
+
+There are usually three types of container-related deliveries.
+
+### Dockerfiles
+
+You will need to ensure that your Dockerfile scripts are security reviewed, that they fetch current versions of software securely from trusted sources, and that you have mechanisms in place to update them as needed if for example it becomes necessary to use new versions of packages due to security errors.
+
+As far as possible you should pull packages from the base distribution itself using the distribution installation tools (e.g., apt, dnf, yum or swupd etc.). Where you cannot do this, you should pull from a trusted source, e.g., an internal build that already has approval. That pull should be via https:// or similar mechanisms and if appropriate also check signatures.
+
+You must list your Dockerfile (in the created/modified components table), along with any software that your Dockerfile copies or installs into your layers (in the unmodified libraries table).
+
+Ensure that your Dockerfile has a license.
+
+### Container Images (layers) pushed to a registry
+
+To get approval for layers (that is, all layers you have generated from your Dockerfile on top of your base/FROM layer), you must have an appropriate SDL or whitelisting in place for each package or component that is present within your layers as well as your own code. You will need to have a process whereby you update your layers whenever there is a security need, and to track the packages within your layers for security problems.
```

Comment:
SDL stands for?

Reply:
"Secure Development Lifecycle". I'll replace it with "security process/approval"...